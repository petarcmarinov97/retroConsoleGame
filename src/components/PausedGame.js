import React, { Component } from 'react';

export default class PausedGame extends Component {
    render() {
        const pauseMsg = "The game is Paused.\n Do you want to continue ?";
        const { defaultStyle } = this.props;

        return <div id="paused" style={defaultStyle}>
            <span>{pauseMsg}</span>
            <div>
                <button id="Yes">Yes</button>
                <button id="No">No</button>
            </div>
        </div>
    }
}
