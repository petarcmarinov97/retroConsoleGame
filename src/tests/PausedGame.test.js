import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import PausedGame from '../components/PausedGame';

describe('PausedGame tests', () => {

    let wrapper;
    const defaultStyle = {
        display: "none"
    }

    beforeEach(() => {
        wrapper = shallow(
            <PausedGame defaultStyle={defaultStyle} />);
    })

    describe('Snapshots', () => {
        it("should render correctly", () => {
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        })
    })
})