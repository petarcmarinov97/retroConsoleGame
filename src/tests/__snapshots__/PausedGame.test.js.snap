// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PausedGame tests Snapshots should render correctly 1`] = `
<div
  id="paused"
  style={
    Object {
      "display": "none",
    }
  }
>
  <span>
    The game is Paused.
 Do you want to continue ?
  </span>
  <div>
    <button
      id="Yes"
    >
      Yes
    </button>
    <button
      id="No"
    >
      No
    </button>
  </div>
</div>
`;
